UnityProject/Assets/Scripts/Core/SecureStuff/SecureMapsSaver.cs

```diff
@@ -68,81 +87,557 @@ public virtual void AddID(string ToAdd)
+			if (attribute.Length > 0)
+			{
+				return false;
+			}
+
+			if (IsSubclassOfRawGeneric(typeof(SyncList<>), Field.FieldType))
+			{
+				return false;
+			}
+
+			if (IsSubclassOfRawGeneric(typeof(SyncDictionary<,>), Field.FieldType))
+			{
+				return false;
+			}
+
+			if (IsSubclassOfRawGeneric(typeof(SyncHashSet<>), Field.FieldType))
+			{
+				return false;
+			}
+
+
+			if (IsSubclassOfRawGeneric(typeof(SyncSortedSet<>), Field.FieldType))
+			{
+				return false;
+			}
+
+			return true;
+		}
+
+		private static bool IsSubclassOfRawGeneric(Type generic, Type toCheck)
+		{
+			while (toCheck != null && toCheck != typeof(object))
+			{
+				var cur = toCheck.IsGenericType ? toCheck.GetGenericTypeDefinition() : toCheck;
+				if (generic == cur)
+				{
+					return true;
+				}
+				toCheck = toCheck.BaseType;
+			}
+			return false;
+		}
+
+		private static bool IsGoodField(FieldInfo Field)
+		{
+			if (Field.IsPrivate || Field.IsAssembly || Field.IsFamily)
+			{
+				var attribute = Field.GetCustomAttributes(typeof(SerializeField), true);
+				if (attribute.Length == 0)
+				{
+					return false;
+				}
+
+				attribute = Field.GetCustomAttributes(typeof(HideInInspector), true);
+				if (attribute.Length > 0)
+				{
+					return false;
+				}
+
+				attribute = Field.GetCustomAttributes(typeof(NaughtyAttributes.ReadOnlyAttribute), true);
+				if (attribute.Length > 0)
+				{
+					return false;
+				}
+
+				attribute = Field.GetCustomAttributes(typeof(PlayModeOnlyAttribute), true);
+				if (attribute.Length > 0)
+				{
+					return false;
+				}
+			}
+			else if (Field.IsPublic)
+			{
+				if (Field.IsNotSerialized)
+				{
+					return false;
+				}
+
+				var attribute = Field.GetCustomAttributes(typeof(PlayModeOnlyAttribute), true);
+				if (attribute.Length > 0)
+				{
+					return false;
+				}
+
+				attribute = Field.GetCustomAttributes(typeof(HideInInspector), true);
+				if (attribute.Length > 0)
+				{
+					return false;
+				}
+
+				attribute = Field.GetCustomAttributes(typeof(NaughtyAttributes.ReadOnlyAttribute), true);
+				if (attribute.Length > 0)
+				{
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+		public static void RecursiveSearchData(HashSet<Component> OnGameObjectComponents,
+			HashSet<GameObject> AllGameObjectOnObject,
+			IPopulateIDRelation IPopulateIDRelation,
+			HashSet<FieldData> FieldDatas,
+			string Prefix,
+			object PrefabInstance,
+			object SpawnedInstance,
+			bool UseInstance = false)
+		{
+			try
+			{
+				var TypeMono = SpawnedInstance.GetType();
+				var coolFields = TypeMono.GetFields(
+					BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic |
+					BindingFlags.FlattenHierarchy
+				).ToList();
+
+				foreach (var Field in coolFields) //Loop through found fields
+				{
+					if (IsGoodField(Field) == false) continue;
+
+					if (Field.Name == "m_dict")
+					{
+						Loggy.LogError("ogggg");
```

Comment:
fasdfjnmasdifonasidfjnasdigfnjasifdgnasdfiugnsdaifujgnsdiufjgnsidfgnsdiufgnaewiurfgiasgfodmsdf

try gauging what I'm saying in this review

Reply:
I'm going to comment out since I need to have this for getting dictionaries working